src/main/java/com/mcmoddev/mmdbot/modules/logging/misc/ScamDetector.java

```diff
@@ -0,0 +1,130 @@
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageUpdateEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+
+import java.awt.Color;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Scam detection system
+ * @author matyrobbrt
+ */
+public class ScamDetector extends ListenerAdapter {
+
+    public static final Gson GSON = new GsonBuilder().disableHtmlEscaping().create();
+    public static final String SCAM_LINKS_DATA_URL = "https://phish.sinking.yachts/v2/all";
+
+    public static final List<String> SCAM_LINKS = Collections.synchronizedList(new ArrayList<>());
+
+    static {
+        new Thread(ScamDetector::setupScamLinks, "Scam link collector").start();
+    }
+
+    @Override
+    public void onGuildMessageReceived(@NotNull final GuildMessageReceivedEvent event) {
+        final var msg = event.getMessage();
+        if (containsScam(msg)) {
+            final var member = msg.getMember();
+            final var guild = msg.getGuild();
+            final var mutedRoleID = MMDBot.getConfig().getRole("muted");
+            final var embed = getLoggingEmbed(msg, "");
+            msg.delete().queue($ -> {
+                getLoggingChannel(guild).sendMessageEmbeds(embed).queue();
+                // TODO once JDA is updated, maybe timeout the user. It seems a better idea
+                guild.addRoleToMember(member, guild.getRoleById(mutedRoleID)).queue();
+            });
+        }
+    }
+
+    @Override
+    public void onGuildMessageUpdate(@NotNull final GuildMessageUpdateEvent event) {
+        final var msg = event.getMessage();
+        if (containsScam(msg)) {
+            final var member = msg.getMember();
+            final var msgChannel = msg.getTextChannel();
+            final var guild = msg.getGuild();
+            final var mutedRoleID = MMDBot.getConfig().getRole("muted");
+            final var embed = getLoggingEmbed(msg, ", by editing an old message");
+            msg.delete().queue($ -> {
+                getLoggingChannel(guild).sendMessageEmbeds(embed).queue();
+                // TODO once JDA is updated, maybe timeout the user. It seems a better idea
+                guild.addRoleToMember(member, guild.getRoleById(mutedRoleID)).queue();
+            });
+        }
+    }
+
+    private static MessageEmbed getLoggingEmbed(final Message message, final String extraDescription) {
+        final var member = message.getMember();
+        return new EmbedBuilder().setTitle("Scam link detected!")
+            .setDescription(String.format("User %s sent a scam link in %s%s! Their message was deleted, and they were muted!", member.getAsMention(),
```

Comment:
A bit of toning down on the exclamation marks. :shipit: 
```suggestion
            .setDescription(String.format("User %s sent a scam link in %s%s. Their message was deleted, and they were muted.", member.getAsMention(),
```
Additionally for posterity and searchability, the tag of the user (`User#getAsTag()`) should also be included.